packages/product-components/src/components/TokenIconSet/TokenIconSet.tsx

```diff
@@ -0,0 +1,68 @@
+import { AssetLogo, useElevation } from '@trezor/components';
+import { getCoingeckoId } from '@suite-common/wallet-config';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+import { TokenInfo } from '@trezor/connect';
+
+import styled, { css } from 'styled-components';
+import { borders, Elevation, mapElevationToBackground, mapElevationToBorder } from '@trezor/theme';
+
+export type TokenIconSetProps = {
+    network: NetworkSymbol;
+    tokens: TokenInfo[];
+    contractAddresses: string[];
+};
+
+const IconContainer = styled.div<{ $length: number }>`
+    width: 26px;
+    justify-content: center;
+    display: flex;
+    align-items: center;
+    ${({ $length }) =>
+        $length > 1 &&
+        css`
+            display: grid;
+            grid-template-columns: repeat(auto-fit, minmax(${$length > 1 ? '1px' : '6px'}, 6px));
+            direction: rtl;
+            justify-items: center;
+        `}
+`;
+
+const TokenIconPlaceholder = styled.div<{ $elevation: Elevation }>`
+    width: 20px;
+    height: 20px;
+    border-radius: ${borders.radii.full};
+    border: 1px solid ${({ theme, $elevation }) => mapElevationToBorder({ theme, $elevation })};
+    background: ${({ theme, $elevation }) => mapElevationToBackground({ theme, $elevation })};
+`;
+
+export const TokenIconSet = ({ network, tokens, contractAddresses }: TokenIconSetProps) => {
+    const { elevation } = useElevation();
+    const { length } = tokens;
+
+    if (length === 0) {
+        return null;
+    }
+
+    const visibleTokens = tokens.slice(0, 3).reverse();
+    //The list of tokens is reversed for styling reasons, so we need to reverse the list of contract addresses as well.
+    const visibleContractAddresses = contractAddresses.slice(0, 3).reverse();
+
+    console.log('VISIBLE TOKENS', visibleTokens);
```

Comment:
:)

Reply:
Thanks, I set up a github hook that won't let me commit console logs anymore 🥲 😅